app/app.go

```diff
@@ -259,3 +271,92 @@ func (a *App) HandleBlock(blk *blockchain.Block) {
 		log.Debug("added blk number %d to chain", blk.BlockNumber)
 	}
 }
+
+// SyncBlockchain updates the local copy of the blockchain by requesting missing
+// blocks from peers. Returns error if we are not connected to any peers.
+func (a *App) SyncBlockchain() error {
```

Comment:
This is a huge function and very difficult to test. Is it possible to break it up a bit?